### PR TITLE
Turning breadcrumb into a partial and improving markup

### PR DESCRIPTION
**Pull request by Guntrisoft**
- improved markup for accessibility
- reduce repetition in templates
- added 'breadcrumb for previous page' functionality for the allocations page

### CODE REVIEW

**Comment by benlovell** [comment]
Would be nice (and reduce some duplication) to wrap the render calls in a helper such as

```ruby
# some helper file
def breadcrumbs(*crumbs)
  render 'shared/breadcrumb', breadcrumb: crumbs
end
```

And then you can call it from the templates like:

```ruby
breadcrumbs({ ... }, { ... }, { ... })
```

**Comment by Guntrisoft** [comment]
@benlovell done!

File: app/views/appointments/new.html.erb [hunk 1]
```diff
@@ -1,7 +1,4 @@
-<ol class="breadcrumb">
-  <li><a href="<%= root_path %>">Home</a></li>
-  <li class="active">Book an appointment</li>
-</ol>
+<%= breadcrumb({ title: "Book an appointment" }) %>
```
Comment (benlovell) [inline app/views/appointments/new.html.erb]: Single quotes.

File: app/views/holidays/edit.html.erb [hunk 2]
```diff
@@ -1,8 +1,8 @@
-<ol class="breadcrumb">
-  <li><a href="<%= root_path %>">Home</a></li>
-  <li><a href="<%= holidays_path %>">Holidays &amp; unavailability</a></li>
-  <li class="active">Edit holiday</li>
-</ol>
+<%= breadcrumb(
+  { title: "Holidays", path: holidays_path },
```
Comment (benlovell) [inline app/views/holidays/edit.html.erb]: Quotes mate.

File: app/views/holidays/index.html.erb [hunk 3]
```diff
@@ -1,7 +1,4 @@
-<ol class="breadcrumb">
-  <li><a href="<%= root_path %>">Home</a></li>
-  <li class="active">Holidays</li>
-</ol>
+<%= breadcrumb( { title: "Holidays" } ) %>
```
Comment (benlovell) [inline app/views/holidays/index.html.erb]: Quotey McQuoteface.

File: app/views/holidays/new.html.erb [hunk 4]
```diff
@@ -1,8 +1,8 @@
-<ol class="breadcrumb">
-  <li><a href="<%= root_path %>">Home</a></li>
-  <li class="active"><a href="<%= holidays_path %>">Holidays</a></li>
-  <li class="active">Create a holiday</li>
-</ol>
+<%= breadcrumb(
+  { title: "Holidays", path: holidays_path },
```
Comment (benlovell) [inline app/views/holidays/new.html.erb]: Don Quixquote.

File: app/views/users/sort.html.erb [hunk 5]
```diff
@@ -1,7 +1,4 @@
-<ol class="breadcrumb">
-  <li><a href="<%= root_path %>">Home</a></li>
-  <li><a href="<%= sort_users_path %>">Guider display order</a></li>
-</ol>
+<%= breadcrumb( { title: "Guider display order" } ) %>
```
Comment (benlovell) [inline app/views/users/sort.html.erb]: You guessed it.

File: app/views/appointments/new.html.erb (hunk 1 above)
Comment (Guntrisoft) [inline app/views/appointments/new.html.erb]: Done

File: app/views/holidays/edit.html.erb (hunk 2 above)
Comment (Guntrisoft) [inline app/views/holidays/edit.html.erb]: Done

File: app/views/holidays/index.html.erb (hunk 3 above)
Comment (Guntrisoft) [inline app/views/holidays/index.html.erb]: Done McDone

File: app/views/holidays/new.html.erb (hunk 4 above)
Comment (Guntrisoft) [inline app/views/holidays/new.html.erb]: Done

File: app/views/users/sort.html.erb (hunk 5 above)
Comment (Guntrisoft) [inline app/views/users/sort.html.erb]: Done

**Comment by Guntrisoft** [comment]
@benbarnett @benlovell all ok?

File: app/helpers/breadcrumb_helper.rb [hunk 6]
```diff
@@ -23,16 +23,20 @@ def previous_page_path
...
     when 'activities#index'
       'My activity'
     when 'my_appointments#show'
       'My appointments'
     when 'company_calendars#show'
-      'Company'
+      'Company appointments'
     else
-      'Search'
+      'Appointment search'
```
Comment (benbarnett) [inline app/helpers/breadcrumb_helper.rb]: I think just "Search" is ok here, it's what it is in the nav + h1

**Pull request merged**